dev/io.openliberty.org.jboss.resteasy.common/src/org/jboss/resteasy/core/ResourceMethodRegistry.java

```diff
@@ -0,0 +1,553 @@
+   {
+      for (ResourceMethod method : resourceClass.getResourceMethods())
+      {
+         processMethod(rf, base, method);
+      }
+      for (ResourceLocator method : resourceClass.getResourceLocators())
+      {
+         processMethod(rf, base, method);
+      }
+   }
+
+   /**
+    * Resteasy 2.x does not properly handle sub-resource and sub-resource locator
+    * endpoints with the same uri.  Resteasy 3.x does handle this properly.  In
+    * assisting customers identify this issue during an upgrade from Resteasy 2 to 3
+    * provides a waring when the situation is found.
+    */
+   public void checkAmbiguousUri()
+   {
+      for (Map.Entry<String, List<ResourceInvoker>> entry : this.root.getBounded().entrySet())
+      {
+         List<ResourceInvoker> values = entry.getValue();
+         if (values.size() > 1) {
+            int locatorCnt = 0;
+            int methodCnt = 0;
+            for(ResourceInvoker rInvoker : values)
+            {
+               if (rInvoker instanceof ResourceLocatorInvoker)
+               {
+                  locatorCnt++;
+               } else if (rInvoker instanceof ResourceMethodInvoker)
+               {
+                  methodCnt++;
+               }
+            }
+            if (methodCnt > 0 && locatorCnt > 0)
+            {
+               StringBuilder sb = new StringBuilder();
+               int cnt = values.size();
+               for (int i=0; i < cnt; i++) {
+                  ResourceInvoker exp = values.get(i);
+                  sb.append(exp.getMethod().getDeclaringClass().getName())
+                          .append(".")
+                          .append(exp.getMethod().getName());
+                  if (i < cnt-1)
+                  {
+                     sb.append(", ");
+                  }
+               }
+               LogMessages.LOGGER.uriAmbiguity(entry.getKey(), sb.toString());
+            }
+         }
+      }
+   }
+
+   // Liberty Change Start
+   @FFDCIgnore(NoClassDefFoundError.class)
+   private void logEndpoint(Method method, String endpoint) {
+
+       LibertyResteasyEndpointLoggingIntrospector introspector = AccessController.doPrivileged(new PrivilegedAction<LibertyResteasyEndpointLoggingIntrospector>(){
+
+           @Override
+           public LibertyResteasyEndpointLoggingIntrospector run() {
+               try {
+               Bundle b = FrameworkUtil.getBundle(ResourceMethodRegistry.class);
+               if(b != null) {
+                   BundleContext bc = b.getBundleContext();
+                   ServiceReference<LibertyResteasyEndpointLoggingIntrospector> sr = bc.getServiceReference(LibertyResteasyEndpointLoggingIntrospector.class);
+                   return (LibertyResteasyEndpointLoggingIntrospector)bc.getService(sr);
+               }
+               } catch (NoClassDefFoundError ncdfe) {
+                   // ignore - return null
+               }
+               return null;
+           }});
+       
```

Comment:
You'll need an early exit to avoid a NPE later in the event that the introspector cannot be found.
```suggestion
       if (introspector == null) return; //optionally log something to trace that introspector couldn't be found
```